File: 1.hs
quicksort [] = []
quicksort (x:xs) = quicksort ys ++ [x] ++ quicksort zs
            where
                ys = [a | a <- xs, a <= x]
                zs = [b | b <- xs, b > x]

quicksort2 xs = case xs of
    [] -> []
    (x:xs) -> quicksort ys ++ [x] ++ quicksort zs
        where
            ys = [a | a <- xs, a <= x]
            zs = [b | b <- xs, b > x]

File: 1.ts
export const quicksort = ([x, ...xs]: number[]) => {
    if (x) {
        const ys = xs.filter(a => a <= x)
        const zs = xs.filter(b => b > x)
        return [...quicksort(ys), x, ...quicksort(zs)]
    } else {
        return []
    }
}